File: 8 kyu - Logical Calculator.py
from functools import reduce

ops = {
    "AND": lambda a, b: a and b,
    "OR": lambda a, b: a or b,
    "XOR": lambda a, b: not b if a else b
}


def logical_calc(array, op):
    return reduce(ops[op], array[1:], array[0])

File: 8 kyu - Logical Calculator.js
function logicalCalc(array, op) {
	switch (op) {
		case 'AND':
			return array.reduce(and);
		case 'OR':
			return array.reduce(or);
		case 'XOR':
			return array.reduce(xor);
	}
}
function and(bool1, bool2) {
	return bool1 && bool2;
}
function or(bool1, bool2) {
	return bool1 || bool2;
}
function xor(bool1, bool2) {
	return bool1 != bool2;
}
